### PR TITLE
[process-agent] Remove remaining process env vars overrides

### PR DESCRIPTION
**Pull request by mbotarro**
### What does this PR do?

This PR removes the remaining process env vars overrides from the `loadEnvVariables` function:
* `DD_PROCESS_AGENT_CONTAINER_SOURCE`, `DD_SCRUB_ARGS`, `DD_STRIP_PROCESS_ARGS` and `DD_CUSTOM_SENSITIVE_WORDS`. These env variables are still supported and corresponding `DD_PROCESS_CONFIG_...` and `DD_PROCESS_AGENT_...` env vars have been created
* `DD_HOSTNAME`, handled by [pkg/config](https://github.com/DataDog/datadog-agent/blob/f0a67887696ed50b0be9f78d7a5f16fee3a7f0aa/pkg/config/config.go#L235)
* `DD_BIND_HOST`, handled by [pkg/config](https://github.com/DataDog/datadog-agent/blob/f0a67887696ed50b0be9f78d7a5f16fee3a7f0aa/pkg/config/config.go#L269)
* `DD_PROXY_HTTPS`, handled by [pkg/config](https://github.com/DataDog/datadog-agent/blob/f0a67887696ed50b0be9f78d7a5f16fee3a7f0aa/pkg/config/config.go#L1048-L1054)

The PR also un-deprecates the `AgentConfig` struct since it has been cleaned up in previous PRs and we can keep using it to share common objects across process-agent checks.

### Motivation

Refactor and improve `process-agent` config logic.

### Describe how to test/QA your changes

Start the `process-agent` with `debug` log level enabled in `datadog.yaml`

```yaml
log_level: 'debug'
```

#### Scrubber Settings

Make sure that scrubber is enabled by default in `process-agent`. Start the agent and make sure that the following log line is printed:

```
Starting process-agent with Scrubber enabled
```

Set
```
export DD_SCRUB_ARGS=true
export DD_CUSTOM_SENSITIVE_WORDS=user,pass*
```

Make sure that the following log lines are printed
```
Starting process-agent with Scrubber enabled
Adding custom sensitives words to Scrubber:  [user pass*]
```

Set 
```
export DD_CUSTOM_SENSITIVE_WORDS='["user","pass*"]'
```

and restart the `process-agent`. Make sure that the expected log lines are still printed.

Start a process with a `user=1234` argument and make sure that it's scrubbed on the Live Process page.

Set 
```
export DD_SCRUB_ARGS=false
```

Make sure that the log line is not printed while starting the agent and that the process that you used in the previous test shows up on the LP page with all its arguments. Rk: you may need to wait a couple of minutes for the processes to be updated.

Unset the previous env vars and repeat the tests with
```
DD_PROCESS_CONFIG_SCRUB_ARGS
DD_PROCESS_CONFIG_CUSTOM_SENSITIVE_WORDS
```
and
````
DD_PROCESS_AGENT_SCRUB_ARGS
DD_PROCESS_AGENT_CUSTOM_SENSITIVE_WORDS
````

#### Strip all processes arguments

When starting the agent, make sure that strip all process arguments it not enabled by default. The following message should not be logged
```
Strip all process arguments enabled
```

Set
```
export DD_STRIP_PROCESS_ARGS=true
```

Restart the agent and make sure that it logs
```
Strip all process arguments enabled
```

Check that all processes on the Live Process page have their args removed. (Rk: you may need to wait a couple of minutes for the processes to be updated).

Set
```
export DD_STRIP_PROCESS_ARGS=false
```

Check that `process-agent` stopped to strip all processes arguments.

Repeat the tests with 
```
DD_PROCESS_CONFIG_STRIP_PROC_ARGUMENTS
DD_PROCESS_AGENT_STRIP_PROC_ARGUMENTS
```

#### Container Source Settings

Set
```
export DD_PROCESS_AGENT_CONTAINER_SOURCE=docker,kubelet
```

Make sure that the following log line is printed
```
Setting container sources to:  [docker,kubelet]
```

Start a container on a host and make sure that it shows up on the LC page.

Repeat the test with `DD_PROCESS_CONFIG_CONTAINER_SOURCE `

#### Hostname

Set
```
export DD_HOSTNAME=<your-hostname>
```

Restart the `process-agent` and make sure that the new host shows up in the logs
```
Starting process-agent for host=<your-hostname> ...
```

Make sure that you can filter processes on the LP page with the `host:your-hostname` tag. (Rk: the core agent needs to be running using `<your-hostname>` as well)

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/process/config/config.go [hunk 1]
```diff
@@ -262,37 +262,22 @@ func getContainerHostType() model.ContainerHostType {
...
-	}
-
-	if v := os.Getenv("DD_PROCESS_ADDITIONAL_ENDPOINTS"); v != "" {
-		endpoints := make(map[string][]string)
-		if err := json.Unmarshal([]byte(v), &endpoints); err != nil {
-			log.Errorf(`Could not parse DD_PROCESS_ADDITIONAL_ENDPOINTS: %v. It must be of the form '{"https://process.agent.datadoghq.com": ["apikey1", ...], ...}'.`, err)
-		} else {
-			config.Datadog.Set("process_config.additional_endpoints", endpoints)
-		}
-	}
```
Comment (mbotarro) [inline pkg/process/config/config.go:294]: `DD_PROCESS_ADDITIONAL_ENDPOINTS ` has been moved to `pkg/config/process.go` on https://github.com/DataDog/datadog-agent/pull/10756

File: pkg/process/config/config.go [hunk 2]
```diff
@@ -262,37 +262,22 @@ func getContainerHostType() model.ContainerHostType {
...
 	for _, variable := range []struct{ env, cfg string }{
-		{"DD_PROCESS_AGENT_CONTAINER_SOURCE", "process_config.container_source"},
-		{"DD_SCRUB_ARGS", "process_config.scrub_args"},
-		{"DD_STRIP_PROCESS_ARGS", "process_config.strip_proc_arguments"},
 		{"DD_ORCHESTRATOR_URL", "orchestrator_explorer.orchestrator_dd_url"},
-		{"DD_HOSTNAME", "hostname"},
-		{"DD_BIND_HOST", "bind_host"},
 		{"HTTPS_PROXY", "proxy.https"},
-		{"DD_PROXY_HTTPS", "proxy.https"},
 	} {
```
Comment (mbotarro) [inline pkg/process/config/config.go:275]: I opted to keep `HTTPS_PROXY` to not risk breaking it for customers that rely on it.

**Review by xornivore — APPROVED** [review]

File: pkg/process/config/config.go (hunk 2 above)
Comment (just-chillin) [reply pkg/process/config/config.go:275]: That's fine I think. The core agent also handles `HTTPS_PROXY` so we should keep it

File: pkg/process/config/config.go [hunk 3]
```diff
@@ -262,37 +262,22 @@ func getContainerHostType() model.ContainerHostType {
...
+// This function is used to handle historic process-agent env vars. New settings should be
+// handled directly in the /pkg/config/process.go file
 func loadEnvVariables() {
 	// The following environment variables will be loaded in the order listed, meaning variables
 	// further down the list may override prior variables.
 	for _, variable := range []struct{ env, cfg string }{
-		{"DD_PROCESS_AGENT_CONTAINER_SOURCE", "process_config.container_source"},
-		{"DD_SCRUB_ARGS", "process_config.scrub_args"},
-		{"DD_STRIP_PROCESS_ARGS", "process_config.strip_proc_arguments"},
 		{"DD_ORCHESTRATOR_URL", "orchestrator_explorer.orchestrator_dd_url"},
```
Comment (just-chillin) [inline pkg/process/config/config.go:273]: You might be able to move this to [here](https://github.com/DataDog/datadog-agent/blob/f0a67887696ed50b0be9f78d7a5f16fee3a7f0aa/pkg/config/config.go#L908), as long as you get permission from containers

File: pkg/process/config/yaml_config.go [hunk 4]
```diff
@@ -26,8 +26,8 @@ func key(pieces ...string) string {
 	return strings.Join(pieces, ".")
 }
 
-// LoadProcessYamlConfig load Process-specific configuration
-func (a *AgentConfig) LoadProcessYamlConfig(path string) error {
+// LoadAgentConfig loads process-agent specific configurations based on the global Config object
+func (a *AgentConfig) LoadAgentConfig(path string) error {
```
Comment (just-chillin) [inline pkg/process/config/yaml_config.go:30]: Nice! 🚀

**Review by just-chillin — APPROVED** [review]

**Comment by just-chillin** [comment]
Looks great! Just a few things

**Review by urseberry — APPROVED** [review]
Docs team approves release notes.

File: pkg/config/process_test.go [hunk 5]
```diff
@@ -290,10 +291,17 @@ func TestEnvVarOverride(t *testing.T) {
 			value:    "1235",
 			expected: 1235,
 		},
+		{
+			key:      "process_config.container_source",
+			env:      "DD_PROCESS_CONFIG_CONTAINER_SOURCE",
+			value:    "[\"docker\",\"kubernetes\"]",
```
[the remark this answers is not in the document]
Comment (mbotarro) [reply pkg/config/process_test.go:297]: Thanks for pointing it out. I normally use raw string literals only when I declare multi-line strings but I think that here we can use it as well. 
Just fixed it!

File: pkg/config/process_test.go [hunk 6]
```diff
@@ -320,6 +328,88 @@ func TestEnvVarOverride(t *testing.T) {
...
+		return cfg.Get(key)
+	}
+}
+
+func TestEnvVarScrubberSettings(t *testing.T) {
+	cfg := setupConf()
+	expectedPrefixes := []string{"DD_", "DD_PROCESS_CONFIG_", "DD_PROCESS_AGENT_"}
+
+	// Scrub enabled/disabled
+	for _, envPrefix := range expectedPrefixes {
```
[the remark this answers is not in the document]
Comment (mbotarro) [reply pkg/config/process_test.go:345]: Fixed! Thanks for pointing it out!

File: pkg/config/process_test.go [hunk 7]
```diff
@@ -320,6 +328,88 @@ func TestEnvVarOverride(t *testing.T) {
...
+				reset := setEnvForTest(e, tc.words)
+				args := cfg.GetStringSlice("process_config.custom_sensitive_words")
+				assert.Equal(t, tc.expected, args)
+				reset()
+			})
+		}
+	}
+
+	// Strip all args
+	for _, envPrefix := range expectedPrefixes {
```
[the remark this answers is not in the document]
Comment (mbotarro) [reply pkg/config/process_test.go:390]: Fixed!

File: pkg/process/config/config.go (hunk 3 above)
Comment (mbotarro) [reply pkg/process/config/config.go:273]: To be honest, I'd prefer to move orchestrator related settings in a follow up PR and keep this one focused on processes ones. What do you think?

**Review by ahmed-mez — APPROVED** [review]

File: pkg/process/config/config.go (hunk 3 above)
Comment (just-chillin) [reply pkg/process/config/config.go:273]: I think that's a valid approach! Let's do that

**Review by just-chillin — APPROVED** [review]

**Pull request merged**